### PR TITLE
Add support for GitHub action release from any branch

### PR DESCRIPTION
**Pull request by sutaakar**
Fixes https://github.com/project-codeflare/instascale/issues/90

### CODE REVIEW

**Comment by sutaakar** [comment]
Also renamed `latest` tag to `stable`

File: .github/workflows/instascale-release.yml [hunk 1]
```diff
@@ -13,10 +13,14 @@ on:
         description: 'Published version of multi-cluster-app-dispatcher to be used as a dependency for InstaScale'
         required: true
         default: 'v0.0.0-dev'
-      is-latest:
-        description: 'Select if the built image should be tagged as latest'
+      is-stable:
```
Comment (KPostOffice) [inline .github/workflows/instascale-release.yml:16]: Why did you change this from `latest` to `stable`

File: .github/workflows/instascale-release.yml [hunk 2]
```diff
@@ -13,10 +13,14 @@ on:
...
+        description: 'Select if the built image should be tagged as stable'
         required: true
         type: boolean
+      branch:
+        description: 'Branch to be used for tag and release'
+        required: true
+        default: 'main'
```
Comment (KPostOffice) [inline .github/workflows/instascale-release.yml:23]: https://github.com/actions/checkout#usage

This is unnecessary. The `checkout` action defaults to the ref that the GitHub action is running for

File: .github/workflows/instascale-release.yml (hunk 1 above)
Comment (anishasthana) [reply .github/workflows/instascale-release.yml:16]: It matches what our release ADR proposed. On more detailed reviewing however... it looks like `dev` is actually the tag that should be updated. We should probably only update the `stable` tag as part of one of the final steps in the codeflare-operator release process. Does that make sense? https://github.com/project-codeflare/adr/blob/main/PCF-ADR-0003-codeflare-release-process.md#images

Comment (sutaakar) [reply .github/workflows/instascale-release.yml:16]: @anishasthana This workflow is triggered for a release, so it should use `stable` tag.
`dev` should be updated in a separate workflow which would be triggered after a push event (workflow doesn't exist yet).

Comment (anishasthana) [reply .github/workflows/instascale-release.yml:16]: That makes sense.

**Review by KPostOffice — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]

**Review by anishasthana — APPROVED** [review]
